### PR TITLE
Refactor logtail, bash, ssh, run to Command; get rid of aws config object

### PR DESCRIPTION
**Pull request by apurvis**
Consolidates https://github.com/lumoslabs/broadside/pull/65 and https://github.com/lumoslabs/broadside/pull/58 with a few other changes.  

- `config.aws` region and credentials are now configured at `config.ecs.region` and `config.ecs.credentials`
- `config.deploy.docker_image` is now `config.default_docker_image`
- validate configuration every time `Broadside.configure` is called and do all do all validation of config up front instead of inside methods that use the config.
- new error classes: `ConfigurationError` (mostly replaces `MissingVariableError` but with more clarity in more situations) and `EcsError` (non fatal ECS response issue)
- get rid of the `super` calls in `EcsDeploy` until the day we have a non ECS deploy
- refactor `check_task_definition!` and `check_service_definition!` to `Target`

### CODE REVIEW

File: lib/broadside/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Broadside
-  VERSION = '2.0.0'
+  VERSION = '3.0.0-prerelease'
```
Comment (apurvis) [inline lib/broadside/version.rb]: @matl33t i noticed that some other more mature projects use this approach to versioning - what's on master is the next tag + `-prerelease`.  I think that's a good strategy; way better than the two other options at least.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 
-Broadside offers a simple command-line interface to perform deployments on ECS. It does not attempt to handle operational tasks like infrastructure setup and configuration, which are better suited for tools like [terraform](https://www.terraform.io/).
+Broadside does _not_ attempt to handle operational tasks like infrastructure setup and configuration, which are better suited to tools like [terraform](https://www.terraform.io/).
 
 ### Things You Can Do With Broadside
 
 - **Deploy a docker image to an ECS cluster and launch a command as an ECS service**
 - **Launch a bash shell directly in the remote docker image** - no messing around with tracking down the server, running `docker ps`, and all the other headaches.
```
Comment (matl33t) [inline README.md]: s/image/container

File: README.md [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
 - **SSH directly onto the server running your image**
```
Comment (matl33t) [inline README.md]: ditto

File: lib/broadside/target.rb [hunk 4]
```diff
@@ -74,14 +85,29 @@ def family
       "#{Broadside.config.application}_#{@name}"
     end
 
-    def to_h
+    def to_hash
```
Comment (matl33t) [inline lib/broadside/target.rb]: https://stackoverflow.com/questions/11182052/to-s-vs-to-str-and-to-i-to-a-to-h-vs-to-int-to-ary-to-hash-in-ruby

Comment (apurvis) [inline lib/broadside/target.rb]: i don't feel strongly about this _but_ the amazon classes all implement `to_hash`; that's the only reason i went with it in broadside.  but i'll change it if you want.

Comment (matl33t) [inline lib/broadside/target.rb]: in https://docs.aws.amazon.com/sdkforruby/api/Aws/ECS/Client.html all of the examples reference `to_h`

Comment (apurvis) [inline lib/broadside/target.rb]: huh well i've definitely been using `to_hash` to investigate stuff, but if that's the case i'll just switch this.

**Comment by matl33t** [comment]
👏

**Pull request merged**